Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MobilityAppBridge.java

```diff
@@ -549,4 +471,211 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
     }
+
+
+    @JavascriptInterface
+    public void addCarouselWithVideo(String stringifyArray, String id) {
+        try{
+            Activity activity = bridgeComponents.getActivity();
+            Context context = bridgeComponents.getContext();
+            JSONObject jsonData = new JSONObject(stringifyArray);
+            int gravity = getGravity(jsonData.getString("gravity"));
+            LinearLayout parentLayout = null;
+            if (activity != null) {
+                parentLayout = activity.findViewById(Integer.parseInt(id));
+            }
+            if (activity == null || parentLayout == null) return;
+            LinearLayout finalParentLayout = parentLayout;
+            activity.runOnUiThread(() -> {
+
+                ViewPager2 viewPager2 = new ViewPager2(context);
+
+                LinearLayout sliderDotsPanel = new LinearLayout(context);
+                LinearLayout.LayoutParams sliderDotsPanelParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+
+                ViewGroup.LayoutParams scrollViewParams = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, finalParentLayout.getHeight() );
+                ScrollView scrollView = new ScrollView(context);
+                scrollView.setLayoutParams(scrollViewParams);
+
+                LinearLayout.LayoutParams linearLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, finalParentLayout.getHeight());
+                LinearLayout linearLayout = new LinearLayout(context);
+                linearLayout.setOrientation(LinearLayout.VERTICAL);
+                linearLayout.setLayoutParams(linearLayoutParams);
+
+                //adding data in array list
+                ArrayList<ViewPagerItem> viewPagerItemArrayList = new ArrayList<>();
+                try {
+                    JSONArray jsonArray = jsonData.getJSONArray("carouselData");
+                    for (int i = 0; i < jsonArray.length(); i++) {
+                        JSONObject jsonObject = jsonArray.getJSONObject(i);
+                        JSONObject imageConfig = jsonObject.getJSONObject("imageConfig");
+                        JSONObject titleConfig = jsonObject.getJSONObject("titleConfig");
+                        JSONObject descriptionConfig = jsonObject.getJSONObject("descriptionConfig");
+                        String contentType = jsonObject.getString("contentType");
+                        JSONObject videoData = jsonObject.getJSONObject("youtubeConfig");
+                        int carouselGravity = jsonObject.getInt("gravity");
+                        int imageID = in.juspay.mobility.app.Utils.getResIdentifier(context,imageConfig.getString("image"), "drawable");
+                        ViewPagerItem viewPagerItem = new ViewPagerItem(imageID, imageConfig, descriptionConfig, titleConfig, contentType, videoData, carouselGravity);
```

Comment:
we should have fallback in case passed decoded string does not have required key, use `opt` for such cases